tests/ui/drop/issue-23611-enum-swap-in-drop.rs

```diff
@@ -240,7 +240,9 @@ pub mod d {
                 let trail = first_avail();
                 let ctr = ((i as u32) * 10_000_000) + (counter as u32);
                 counter += 1;
+                //~^ WARN creating a mutable reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.